src/Protocol.cpp

```diff
@@ -606,3 +600,100 @@ PeriodicUpdate protocol::parseE4Output(uint8_t const* buffer, int bufferSize)
     result.filter_state = state;
     return result;
 }
+
+PeriodicUpdate protocol::parseE5Output(uint8_t const* buffer, int bufferSize)
+{
+    uint8_t const* end = buffer + bufferSize;
+
+    uint8_t const* cursor = buffer;
+    uint32_t time_ms;
+    cursor = endianness::decode<uint32_t>(buffer, time_ms, end);
+    uint8_t filterFlags = cursor[0];
+    ++cursor;
+    int8_t board_temperature_C = *reinterpret_cast<int8_t const*>(cursor);
+    ++cursor;
+
+    float values[10];
```

Comment:
I know this is a fairly simple code, but it repeats and we have to type the amount of data twice (in array declaration and in the for loop). Maybe it would be more safe this way:

```cpp
template <typename T, std::size_t n>
uint8_t* array_decode(uint8_t const* cursor, uint8_t const* end,  std::array<T, N> &array)
{
  for(std::size_t i = 0; i < N; i++) {
    cursor = endianness::decode<T>(cursor, array[i], end); 
  }
  return cursor;
}
```
and then use 
```cpp
std::array<float, 10> values;
cursor = array_decode(cursor, end, values);
```

Also why read attitude, angular and linear velocities all inside the values array, when you  created a separated array for the other variables?

Reply:
I was also bothered by the duplication of the array size and of the loop size ... it does smell like something that could backfire. I'll change it.